index.rst

```diff
@@ -1,25 +1,370 @@
+
+The Engineering Facilities Database
+-----------------------------------
+
+DDS requires running complex extra software (ts_salkafka, an in-house package) in order to copy all messages from DDS to Kafka, so we can ingest them into the Engineering Facilities Database (EFD).
+If we use Kafka then we can eliminate this extra software.
+
+Performance
+===========
+
+I measured performance with two ``MTM1M3`` topics: the ``summaryState event``, which is one of our smallest topics, and the ``forceActuatorData`` telemetry topic, which is one of our largest topics.
+All timings were measured on summit computer ``azar02``.
+
+======  =================   ============    ============    =======     =======     =======     =======     =============
+System  Topic               Write Speed     Read Speed      Latency     Latency     Latency     Latency     Lost Messages
+                            messages/sec    messages/sec    mean ms     sdev ms     min ms      max ms
+======  =================   ============    ============    =======     =======     =======     =======     =============
+DDS     summaryState        16,848          16,887           0           1           0           5              0
+------  -----------------   ------------    ------------    -------     -------     -------     -------     -------------
+DDS     forceActuatorData    1,875           1,512           1           0           1           1          2,173
+------  -----------------   ------------    ------------    -------     -------     -------     -------     -------------
+Kafka   summaryState         2,084           2,086           1           1           1          23              0
+------  -----------------   ------------    ------------    -------     -------     -------     -------     -------------
+Kafka   forceActuatorData    1,605           1,607           2           1           1          23              0
+======  =================   ============    ============    =======     =======     =======     =======     =============
+
+Kafka performance appears to be fine.
+
+Our highest message rate is 20 Hz for tracking commands and some telemetry, including MTM1M3 forceActuatorData.
+The tracking commands will have to be sent at least 25 ms or so in advance in order to deal with the Kafka latency.
+We need more than this to deal with the variability in Python performance, since our axis controllers use Python CSCs to handle the SAL messages and control low-level controllers.
+
+Measurement Details
+-------------------
+
+* Due to the way I took the measurements, measured read speed cannot be significantly higher than write speed.
+  In most cases the reader was able to keep up with the writer, but in one case it lagged enough to lose messages.
+
+* All timings used 10,000 messages except DDS forceActuatorData, where wrote 20,000 messages and read 10,000.
+  I had to write more messages than I read, because some messages were lost.
```

Comment:
Can you give an estimate of % message lost? Maybe add this to the table?

Reply:
I think it's not actually useful. The question is only whether the salobj reader can keep up. As long as it is not DDS or Kafka losing messages it's not a statement about reliability. I have rewritten this section to clarify that.